aiven/resource_flink_table.go

```diff
@@ -0,0 +1,219 @@
+		ForceNew:    true,
+	},
+}
+
+func resourceFlinkTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceFlinkTableCreate,
+		ReadContext:   resourceFlinkTableRead,
+		DeleteContext: resourceFlinkTableDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceFlinkTableState,
+		},
+
+		Schema: aivenFlinkTableSchema,
+	}
+}
+
+func resourceFlinkTableRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project, serviceName, tableId := splitResourceID3(d.Id())
+
+	r, err := client.FlinkTables.Get(project, serviceName, aiven.GetFlinkTableRequest{TableId: tableId})
+	if err != nil {
+		return diag.FromErr(resourceReadHandleNotFound(err, d))
+	}
+
+	if err := d.Set("project", project); err != nil {
+		return diag.Errorf("error setting Flink Tables `project` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("service_name", serviceName); err != nil {
+		return diag.Errorf("error setting Flink Tables `service_name` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("integration_id", r.IntegrationId); err != nil {
+		return diag.Errorf("error setting Flink Tables `integration_id` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("table_id", r.TableId); err != nil {
+		return diag.Errorf("error setting Flink Tables `table_id` for resource %s: %s", d.Id(), err)
+	}
+	if err := d.Set("table_name", r.TableName); err != nil {
+		return diag.Errorf("error setting Flink Tables `table_name` for resource %s: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceFlinkTableState(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	di := resourceFlinkTableRead(ctx, d, m)
+	if di.HasError() {
+		return nil, fmt.Errorf("cannot get flink table: %v", di)
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceFlinkTableCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+	integrationId := d.Get("integration_id").(string)
+	jdbcTable := d.Get("jdbc_table").(string)
+	kafkaTopic := d.Get("kafka_topic").(string)
+	likeOptions := d.Get("like_options").(string)
+	tableName := d.Get("table_name").(string)
+	partitionedBy := d.Get("partitioned_by").(string)
+	schemaSQL := d.Get("schema_sql").(string)
+
+	createRequest := aiven.CreateFlinkTableRequest{
+		IntegrationId: integrationId,
+		JDBCTable:     jdbcTable,
+		KafkaTopic:    kafkaTopic,
+		LikeOptions:   likeOptions,
+		Name:          tableName,
+		PartitionedBy: partitionedBy,
+		SchemaSQL:     schemaSQL,
+	}
+
+	conf := &resource.StateChangeConf{
+		Pending: []string{"CREATING"},
+		Target:  []string{"CREATED"},
+		Refresh: func() (interface{}, string, error) {
+			r, err := client.FlinkTables.Create(project, serviceName, createRequest)
+			if err != nil {
+				if aiven.IsNotFound(err) {
```

Comment:
What shall we do if a user sets a wrong `integration_id`? I always find silencing 404 responses quite dangerous during the creation of a resource.  Is it indispensable here?

Reply:
fixed service integration waiter; this should now work because the service integration was fixed.

Reply:
removed the waiter from flink table creation